### PR TITLE
store & load comments using Datastore instead of using an ArrayList

### PR DESCRIPTION
**Pull request by ashleygranitto**
I must have accidentally duplicated some of my commits--only the "store user comments via datastore" and "load user comments via datastore" commits were intended for this branch

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/FormServlet.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Retrieve newest user comment and store it as a Comment Entity
+    String userComment = request.getParameter("user-comment");
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("text", userComment);
+
+    // Store the Comment Entity in Datastore 
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```
Comment (mmallett) [inline portfolio/src/main/java/com/google/sps/servlets/FormServlet.java]: How does the servlet know which database and table to connect to? Is it stored in some config file?

File: portfolio/src/main/java/com/google/sps/servlets/FormServlet.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that returns nametag text. */
```
Comment (mmallett) [inline portfolio/src/main/java/com/google/sps/servlets/FormServlet.java]: update this comment

Comment (ashleygranitto) [reply portfolio/src/main/java/com/google/sps/servlets/FormServlet.java]: updated

File: portfolio/src/main/java/com/google/sps/servlets/FormServlet.java (hunk 1 above)
Comment (ashleygranitto) [reply portfolio/src/main/java/com/google/sps/servlets/FormServlet.java]: I accidentally forgot to include a commit of my updated pom.xml file--I included a dependency to Google AppEngine here. Regarding specific tables, to my understanding Datastore just keeps track of user-created groups of Entities. Please let me know if you want me to include more specific implementation information within my comments or take any other steps!

**Comment by ashleygranitto** [comment]
I apologize I again somehow duplicated my commits on both branches--I will work to avoid this in the future. The commits that were intended for this branch are 'include Google AppEngine dependency' and 'update documentation comment'

File: portfolio/src/main/java/com/google/sps/servlets/FormServlet.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    datastore.put(commentEntity);
+
+    // Redirect back to the HTML page
+    response.sendRedirect("/comment.html");
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Comment");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```
Comment (win-chen) [inline portfolio/src/main/java/com/google/sps/servlets/FormServlet.java]: Should this datastore be a private property on the class? Or is it important that it gets created at every get and post?

Comment (ashleygranitto) [reply portfolio/src/main/java/com/google/sps/servlets/FormServlet.java]: I originally based my implementation off of the tutorial's example where they created it at every get and post, but I made it private and the functionality was not affected & I could not find any justification for needing it at every get and post within the tutorial instructions. I just pushed a version where datastore is private--please let me know if you meant something different by private property.

File: portfolio/src/main/java/com/google/sps/servlets/FormServlet.java (hunk 1 above)
Comment (mmallett) [reply portfolio/src/main/java/com/google/sps/servlets/FormServlet.java]: That sounds good, was just curious

File: portfolio/src/main/java/com/google/sps/servlets/FormServlet.java (hunk 3 above)
Comment (mmallett) [reply portfolio/src/main/java/com/google/sps/servlets/FormServlet.java]: It depends on how expensive it is to build, and if there is any configuration baked into it. A typical database connection (like MySQL) would actually be managed by the app framework context, it's a global concern. The app framework manages a pool of connections, and injects a connection into the servlet when it receives a request.

Though, MySQL connections need to be maintained, and are much more expensive. I don't know much of how DataStore works, but it looks to me to be a stateless, HTTP based API. So the connection overhead is basically non-existent.

**Review by mmallett — APPROVED** [review]

**Pull request merged**